components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -37,7 +37,9 @@
         function submitCredentials () {
             var userName = document.getElementById("username");
             userName.value = userName.value.trim();
-            document.getElementById("loginForm").submit();
+            if(userName.value != ''){
```

Comment:
better to refer [1] for JavaScript empty checks.  
[1]https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in